lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -650,8 +650,10 @@ impl ShardReplicaSet {
                             .await?;
                     }
                     ReplicaState::Initializing => {
-                        self.set_local(local_shard, Some(ReplicaState::Initializing))
+                        // Same as `Active`, we report a failure to consensus
+                        self.set_local(local_shard, Some(ReplicaState::Active))
```

Comment:
I don't think we can set Active local state here, even if the Active and Initializing will act the same.

General rule here is that we only change state of shards if there is a consensus operation saying to do so. Otherwise there is a risk of having de-sync in shard states between nodes.


Consider the following scenario:

- Node A of a newest version receives Snapshot with `ReplicaState::Initializing` and changes it to Active.
- Node B with an older version still thinks that Node A have shard in Initializing state.
- Since Node A have local state of Active, it will never trigger consensus to change the state
- Node B will think about node A as stuck in Initializing state indefinitely,  even after update to a newer version.


Luckily it should not be a big deal to have `Initializing` state, as we have a procedure to convert Initializing into active if it happens to be a local one

Reply:
I see, thank you for your explication. I will revert this commit because there won't be the case to notify peer failure.